server/src/main/java/org/opensearch/common/util/concurrent/ThreadContext.java

```diff
@@ -246,6 +256,14 @@ public StoredContext newStoredContext(boolean preserveResponseHeaders, Collectio
         }
         // this is the context when this method returns
         final ThreadContextStruct newContext = threadLocal.get();
+
+        if (newContext.transientHeaders.containsKey(CURRENT_SPAN)) {
```

Comment:
What is the purpose of this code block? It gets and sets the same value?

Reply:
We are actually doing a deep copy here. Otherwise two threads will share the same atomic reference instance and mutate it.

Reply:
:+1: I see now, thank you